trade-module/src/main/java/io/edpn/backend/trade/application/service/RequestSystemCoordinatesService.java

```diff
@@ -33,18 +33,18 @@ public boolean isApplicable(System system) {
     }
 
     @Override
-    public void request(System system) {
+    public synchronized void request(System system) {
```

Comment:
I'm guessing synchronization is here to prevent a check-then act?

It does the job (as long as this is the only method creating those requests), but in principle I would much prefer if there was a "createIfNotExists" kind of transactional method for system coordinate requests. It could return an `Optional` of request, and `ifPresent()` used to send the message.

Reply:
this still introduces racing conditions does it not?